src/main/java/com/frameworkium/core/ui/pages/Visibility.java

```diff
@@ -58,8 +59,20 @@ public Visibility(Wait<WebDriver> wait, JavascriptExecutor driver) {
      */
     public void waitForAnnotatedElementVisibility(Object pageObject) {
 
-        Field[] allFields = pageObject.getClass().getDeclaredFields();
-        Arrays.stream(allFields)
+        Class <? extends Object> clazz = pageObject.getClass();
+        // Get the declared fields from the current class
+        final List<Field> allFields = new ArrayList<Field>(Arrays.asList(clazz.getDeclaredFields()));
+
+        // Get any declared fields from super classes
+        // i.e. when a page object extends a custom class which itself extends HtmlElement
+        for(clazz = clazz.getSuperclass(); 
```

Comment:
What happens if this super class has another one? Should this be recursive until you hit `Object` (or `null`) or `BasePage` or `HtmlElement`?

Reply:
Yes, it is recursive